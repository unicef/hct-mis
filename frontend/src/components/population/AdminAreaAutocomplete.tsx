import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import get from 'lodash/get';
import { InputAdornment } from '@material-ui/core';
import RoomRoundedIcon from '@material-ui/icons/RoomRounded';
import CircularProgress from '@material-ui/core/CircularProgress';
import Autocomplete from '@material-ui/lab/Autocomplete';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasLazyQuery,
} from '../../__generated__/graphql';
import { useBusinessArea } from '../../hooks/useBusinessArea';

const StyledAutocomplete = styled(Autocomplete)`
  width: ${(props) => (props.fullWidth ? '100%' : '232px')}
    .MuiFormControl-marginDense {
    margin-top: 4px;
  }
`;

export function AdminAreasAutocomplete({
  disabled,
  fullWidth,
  onFilterChange,
  name,
}: {
  disabled?;
  fullWidth?: boolean;
  onFilterChange;
  name: string;
}): React.ReactElement {
  const [open, setOpen] = useState(false);
  const [inputValue, onInputTextChange] = useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const businessArea = useBusinessArea();
  const [loadAdminAreas, { data, loading }] = useAllAdminAreasLazyQuery({
    variables: {
      first: 50,
      title: debouncedInputText,
      businessArea,
      level: 2,
    },
  });
  useEffect(() => {
    loadAdminAreas();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [debouncedInputText]);

  const onChangeMiddleware = (e, selectedValue): void => {
    onFilterChange((filters) => ({
      ...filters,
      [name]: selectedValue || undefined,
    }));
  };
  return (
    <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
      fullWidth={fullWidth}
      open={open}
      filterOptions={(options1) => options1}
      onChange={onChangeMiddleware}
      onOpen={() => {
        setOpen(true);
      }}
      onClose={(e, reason) => {
        setOpen(false);
        if (reason === 'select-option') return;
        onInputTextChange('');
      }}
      getOptionSelected={(option, value1) => {
        return value1?.node?.id === option.node.id;
      }}
      getOptionLabel={(option) => {
        if (!option.node) {
          return '';
        }
        return `${option.node.title}`;
      }}
      disabled={disabled}
      options={get(data, 'allAdminAreas.edges', [])}
      loading={loading}
      renderInput={(params) => (
        <TextField
          {...params}
          label='Admin Level 2'
          variant='outlined'
          margin='dense'
          value={inputValue}
          onChange={(e) => onInputTextChange(e.target.value)}
          InputProps={{
            ...params.InputProps,
            startAdornment: (
              <InputAdornment position='start'>
                <RoomRoundedIcon style={{ color: '#5f6368' }} />
              </InputAdornment>
            ),
            endAdornment: (
              <>
                {loading ? (
                  <CircularProgress color='inherit' size={20} />
                ) : null}
                {params.InputProps.endAdornment}
              </>
            ),
          }}
        />
      )}
    />
  );
}
