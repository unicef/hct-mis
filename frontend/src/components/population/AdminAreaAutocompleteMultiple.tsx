import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import get from 'lodash/get';
import { Box } from '@material-ui/core';
import Autocomplete from '@material-ui/lab/Autocomplete';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasQuery,
} from '../../__generated__/graphql';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { LoadingComponent } from '../LoadingComponent';
import { FieldLabel } from '../FieldLabel';

const StyledAutocomplete = styled(Autocomplete)`
  width: 232px;
  .MuiFormControl-marginDense {
    margin-top: 4px;
  }
`;

export function AdminAreaAutocompleteMultiple({
  value,
  onChange,
  disabled,
}: {
  value;
  onChange;
  disabled?;
}): React.ReactElement {
  const [open, setOpen] = React.useState(false);
  const [inputValue, setInputTextChange] = React.useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const [newValue, setNewValue] = useState([]);
  const businessArea = useBusinessArea();
  const { data, loading } = useAllAdminAreasQuery({
    variables: {
      first: 100,
      title: debouncedInputText,
      businessArea,
    },
  });
  useEffect(() => {
    setNewValue(value);
  }, [data, value]);
  useEffect(() => {
    setInputTextChange('');
  }, [value]);

  if (loading) return <LoadingComponent />;
  if (!data) return null;
  return (
    <Box display='flex' flexDirection='column'>
      <FieldLabel>Administrative Level 2</FieldLabel>
      <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
        open={open}
        multiple
        fullWidth
        filterOptions={(options1) => options1}
        onChange={onChange}
        value={newValue}
        onOpen={() => {
          setOpen(true);
        }}
        onClose={() => {
          setOpen(false);
        }}
        getOptionSelected={(option, value1) => {
          return value1?.node?.id === option.node.id;
        }}
        getOptionLabel={(option) => {
          if (!option.node) {
            return '';
          }
          return `${option.node.title}`;
        }}
        disabled={disabled}
        options={get(data, 'allAdminAreas.edges', [])}
        loading={loading}
        renderInput={(params) => {
          return (
            <TextField
              {...params}
              inputProps={{
                ...params.inputProps,
                value: inputValue,
              }}
              placeholder={
                newValue.length > 0 ? null : 'Administrative Level 2'
              }
              variant='outlined'
              margin='dense'
              value={inputValue}
              onChange={(e) => setInputTextChange(e.target.value)}
            />
          );
        }}
      />
    </Box>
  );
}
