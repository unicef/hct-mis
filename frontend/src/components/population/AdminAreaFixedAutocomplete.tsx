import { InputAdornment } from '@mui/material';
import CircularProgress from '@mui/material/CircularProgress';
import RoomRoundedIcon from '@mui/icons-material/RoomRounded';
import Autocomplete from '@mui/lab/Autocomplete';
import get from 'lodash/get';
import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasQuery,
} from '../../__generated__/graphql';

const StyledAutocomplete = styled(Autocomplete)`
  .MuiFormControl-marginDense {
    //margin-top: 4px;
  }
`;

export function AdminAreaFixedAutocomplete({
  value,
  onChange,
  disabled,
}: {
  value;
  onChange;
  disabled?;
}): React.ReactElement {
  const { t } = useTranslation();
  const [open, setOpen] = React.useState(false);
  const [inputValue, onInputTextChange] = React.useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const [newValue, setNewValue] = useState(null);
  const businessArea = useBusinessArea();
  const { data, loading } = useAllAdminAreasQuery({
    variables: {
      name: debouncedInputText,
      businessArea,
      first: 50,
      level: 2,
    },
  });

  useEffect(() => {
    if (data) {
      setNewValue(
        typeof value === 'string'
          ? data.allAdminAreas.edges.find((item) => item.node.name === value)
          : value,
      );
    } else {
      // setNewValue(value);
    }
    // onInputTextChange('');
  }, [data, value]);
  const onChangeMiddleware = (e, selectedValue, reason): void => {
    onInputTextChange(selectedValue?.node?.name);
    onChange(e, selectedValue, reason);
  };
  return (
    <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
      open={open}
      filterOptions={(options1) => options1}
      onChange={onChangeMiddleware}
      value={newValue}
      onOpen={() => {
        setOpen(true);
      }}
      onClose={(e, reason) => {
        setOpen(false);
        if (value || reason === 'select-option') return;
        onInputTextChange(null);
      }}
      getOptionSelected={(option, selectedValue) => {
        return selectedValue?.node?.id === option.node.id;
      }}
      getOptionLabel={(option) => {
        if (!option.node) {
          return '';
        }
        return `${option.node.name}`;
      }}
      disabled={disabled}
      options={get(data, 'allAdminAreas.edges', [])}
      loading={loading}
      renderInput={(params) => (
        <TextField
          {...params}
          label={t('Administrative Level 2')}
          variant='outlined'
          margin='dense'
          value={inputValue}
          onChange={(e) => onInputTextChange(e.target.value)}
          InputProps={{
            ...params.InputProps,
            startAdornment: (
              <InputAdornment position='start'>
                <RoomRoundedIcon style={{ color: '#5f6368' }} />
              </InputAdornment>
            ),
            endAdornment: (
              <>
                {loading ? (
                  <CircularProgress color='inherit' size={20} />
                ) : null}
                {params.InputProps.endAdornment}
              </>
            ),
          }}
        />
      )}
    />
  );
}
