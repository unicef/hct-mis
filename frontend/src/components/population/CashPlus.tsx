import React from 'react';
import styled from 'styled-components';
import { Typography, Paper, Grid } from '@material-ui/core';
import { LabelizedField } from '../LabelizedField';

const Overview = styled(Paper)`
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
  margin-bottom: ${({ theme }) => theme.spacing(6)}px;
`;

const Title = styled.div`
  width: 100%;
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

export function CashPlus({ individual }): React.ReactElement {
  const { enrolledInNutritionProgramme, administrationOfRutf } = individual;
  return (
    <Overview>
      <Title>
        <Typography variant='h6'>Cash+</Typography>
      </Title>
      <Grid container spacing={6}>
        <Grid item xs={4}>
          <LabelizedField
            label='Enrolled in nutrition programme'
            value={enrolledInNutritionProgramme ? 'YES' : 'NO'}
          />
        </Grid>
        <Grid item xs={4}>
          <LabelizedField
            label='Administratiion of rutf'
            value={administrationOfRutf ? 'YES' : 'NO'}
          />
        </Grid>
      </Grid>
    </Overview>
  );
}
