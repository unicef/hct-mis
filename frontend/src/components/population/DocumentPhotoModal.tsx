import React, { useState } from 'react';
import styled from 'styled-components';
import { Button, DialogContent, DialogTitle, Box } from '@material-ui/core';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { BlackLink } from '../BlackLink';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const StyledImage = styled.img`
  max-width: 100%;
  max-height: 100%;
`;

interface DocumentPhotoModalProps {
  photo: string;
  documentId: string;
}

export const DocumentPhotoModal = ({
  photo,
  documentId,
}: DocumentPhotoModalProps): React.ReactElement => {
  const [dialogOpen, setDialogOpen] = useState(false);

  return (
    <>
      <BlackLink
        onClick={() => {
          setDialogOpen(true);
        }}
      >
        {documentId}
      </BlackLink>
      <Dialog
        open={dialogOpen}
        onClose={() => setDialogOpen(false)}
        aria-labelledby='form-dialog-title'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            Document&apos;s Photo
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <Box p={3}>
            <StyledImage alt='document' src={photo} />
          </Box>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setDialogOpen(false)}>CANCEL</Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
