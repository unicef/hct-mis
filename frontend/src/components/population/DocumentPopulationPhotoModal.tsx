import { Box, Button, DialogContent } from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import {
  IndividualNode,
  useIndividualPhotosLazyQuery,
} from '../../__generated__/graphql';
import { PhotoModalHeader } from '../PhotoModal/PhotoModalHeader';
import {
  DialogFooter,
  StyledImage,
  StyledLink,
} from './IndividualFlexFieldPhotoModal';

interface DocumentPopulationPhotoModalProps {
  individual: IndividualNode;
  documentNumber: string;
  documentId: string;
}

export const DocumentPopulationPhotoModal = ({
  individual,
  documentNumber,
  documentId,
}: DocumentPopulationPhotoModalProps): React.ReactElement => {
  const [turnAngle, setTurnAngle] = useState(90);
  const { t } = useTranslation();
  const [dialogOpen, setDialogOpen] = useState(false);
  const [getPhotos, { data }] = useIndividualPhotosLazyQuery({
    variables: { id: individual?.id },
    fetchPolicy: 'network-only',
  });
  const documentWithPhoto = data?.individual?.documents?.edges?.find(
    (el) => el.node.id === documentId,
  );

  return (
    <>
      <StyledLink
        onClick={() => {
          setDialogOpen(true);
          getPhotos();
        }}
      >
        {documentNumber}
      </StyledLink>
      <Dialog
        open={dialogOpen}
        onClose={() => setDialogOpen(false)}
        aria-labelledby='form-dialog-title'
      >
        <PhotoModalHeader
          turnAngle={turnAngle}
          setTurnAngle={setTurnAngle}
          title="Document's Photo"
        />
        <DialogContent>
          <Box p={3}>
            <StyledImage
              id='modalImg'
              alt='document'
              src={documentWithPhoto?.node?.photo}
            />
          </Box>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setDialogOpen(false)}>{t('CANCEL')}</Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
