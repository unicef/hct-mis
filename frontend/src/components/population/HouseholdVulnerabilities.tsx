import React from 'react';
import styled from 'styled-components';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import { Typography } from '@material-ui/core';
import { LabelizedField } from '../LabelizedField';
import { HouseholdDetailedFragment } from '../../__generated__/graphql';

const Overview = styled(Paper)`
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
  margin-top: 20px;
  &:first-child {
    margin-top: 0px;
  }
`;

const Title = styled.div`
  width: 100%;
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

interface HouseholdVulnerabilities {
  household: HouseholdDetailedFragment;
}

export function HouseholdVulnerabilities({ household }): React.ReactElement {
  const fields = [];
  // eslint-disable-next-line guard-for-in,no-restricted-syntax
  for (const flexFieldKey in household.flexFields) {
    fields.push(
      <LabelizedField
        key={flexFieldKey}
        label={flexFieldKey}
        value={household.flexFields[flexFieldKey]}
      />,
    );
  }
  return (
    <div>
      <Overview>
        <Title>
          <Typography variant='h6'>Vulnerabilities</Typography>
        </Title>
        <Grid container spacing={6}>
          <Grid item xs={4}>
            {fields}
          </Grid>
        </Grid>
      </Overview>
    </div>
  );
}
