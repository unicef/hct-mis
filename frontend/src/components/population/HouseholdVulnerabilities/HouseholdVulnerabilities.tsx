import { Typography } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { useArrayToDict } from '../../../hooks/useArrayToDict';
import {
  HouseholdNode,
  AllHouseholdsFlexFieldsAttributesQuery,
} from '../../../__generated__/graphql';
import { LabelizedField } from '../../core/LabelizedField';
import { Title } from '../../core/Title';
import { HouseholdFlexFieldPhotoModal } from '../HouseholdFlexFieldPhotoModal';

const Overview = styled(Paper)`
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
  margin-top: 20px;
  &:first-child {
    margin-top: 0px;
  }
`;

interface HouseholdVulnerabilitiesProps {
  household: HouseholdNode;
  flexFieldsData: AllHouseholdsFlexFieldsAttributesQuery;
}

export function HouseholdVulnerabilities({
  household,
  flexFieldsData,
}: HouseholdVulnerabilitiesProps): React.ReactElement {
  const { t } = useTranslation();

  const flexAttributesDict = useArrayToDict(
    flexFieldsData?.allHouseholdsFlexFieldsAttributes,
    'name',
    '*',
  );

  const fields = Object.entries(household.flexFields || {}).map(
    ([key, value]: [string, string | string[]]) => {
      if (flexAttributesDict[key]?.type === 'IMAGE') {
        return (
          <LabelizedField
            key={key}
            label={key.replaceAll('_i_f', '').replace(/_/g, ' ')}
          >
            <HouseholdFlexFieldPhotoModal field={flexAttributesDict[key]} />
          </LabelizedField>
        );
      }
      if (
        flexAttributesDict[key]?.type === 'SELECT_MANY' ||
        flexAttributesDict[key]?.type === 'SELECT_ONE'
      ) {
        let newValue =
          flexAttributesDict[key].choices.find((item) => item.value === value)
            ?.labelEn || '-';
        if (value instanceof Array) {
          newValue = value
            .map(
              (choice) =>
                flexAttributesDict[key].choices.find(
                  (item) => item.value === choice,
                )?.labelEn || '-',
            )
            .join(', ');
        }
        return (
          <LabelizedField
            key={key}
            label={key.replaceAll('_i_f', '').replace(/_/g, ' ')}
            value={newValue}
          />
        );
      }
      return (
        <LabelizedField
          key={key}
          label={key.replaceAll('_h_f', '').replace(/_/g, ' ')}
          value={value}
        />
      );
    },
  );

  return (
    <div>
      <Overview>
        <Title>
          <Typography variant='h6'>{t('Vulnerabilities')}</Typography>
        </Title>
        <Grid container spacing={6}>
          {fields.map((field) => (
            <Grid item xs={4}>
              {field}
            </Grid>
          ))}
        </Grid>
      </Overview>
    </div>
  );
}
