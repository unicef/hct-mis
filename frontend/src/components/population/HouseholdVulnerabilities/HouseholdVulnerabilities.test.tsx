import * as React from 'react';
import { fakeAllHouseholdsFlexFieldsAttributes } from '../../../../fixtures/core/fakeAllHouseholdsFlexFieldsAttributes';
import { fakeHousehold } from '../../../../fixtures/population/fakeHousehold';
import { render } from '../../../testUtils/testUtils';
import { HouseholdVulnerabilities } from './HouseholdVulnerabilities';

describe('components/population/HouseholdVulnerabilities', () => {
  it('should render', () => {
    const { container } = render(
      <HouseholdVulnerabilities
        household={fakeHousehold}
        flexFieldsData={fakeAllHouseholdsFlexFieldsAttributes}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
