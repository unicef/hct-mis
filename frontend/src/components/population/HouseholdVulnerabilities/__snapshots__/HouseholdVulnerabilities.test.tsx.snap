// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/population/HouseholdVulnerabilities should render 1`] = `
<div>
  <div>
    <div
      class="MuiPaper-root sc-htoDjs bshDgT MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="sc-htpNat hUqbZq"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Vulnerabilities
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div>
            <span
              class="sc-bdVaJa kwsjSG"
              color="textSecondary"
            >
              months displaced
            </span>
            <div
              data-cy="label-months displaced"
            >
              <span
                class="sc-bwzfXH eVLibZ"
                color="textSecondary"
              >
                12
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div>
            <span
              class="sc-bdVaJa kwsjSG"
              color="textSecondary"
            >
              difficulty breathing h f
            </span>
            <div
              data-cy="label-difficulty breathing h f"
            >
              <span
                class="sc-bwzfXH eVLibZ"
                color="textSecondary"
              >
                Yes
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
